### DISCUSSION TITLE
v2.0.35b4

### DISCUSSION THREAD

**Discussion by DTTerastar** [ESPresense/ESPresense#358] [Announcements]
More efficient looping
Fixes to room assistant detection

<hr /><em>This discussion was created from the release <a href='https://github.com/ESPresense/ESPresense/releases/tag/v2.0.35b4'>v2.0.35b4</a>.</em>

**Comment by harphere** [comment]
This is looking okay so far. Distances are back to a normal look number now.

**Comment by harphere** [comment]
Having tested for the day and continuing tomorrow, the reliability I find is much better than before. I have an Android and iOS phone using HA app and fingerprint and both are broadcasting and staying in their rooms without drifting too much. Good work! I will continue to watch tomorrow.

**Reply by DTTerastar** [reply]
I'm going to release it once I figure out why one of my devices can't auto update.

**Reply by harphere** [reply]
I've observed the following behavior before but couldn't articulate it before. One of the device ids for my wife's iPhone 12 Pro is apple:1005:9-12, this becomes the id sometimes randomly sometimes after a reboot of the phone. This afternoon, while she was out of the house, this device id attached itself to my bedroom ESP32 which would have thrown off an automation had I had one. Because these ids rotate, the one I actually want is away but the one that is not currently in rotation becomes 'home'. The 'native' fingerprint apple:iphone13-3 never appears and hasn't appeared in a long time. It is hard to keep track of the rotated ids when one gets replaced with another. How best to handle these scenarios? Is it to set up the query boxes? I've included all possible device ids I found in the query but that doesn't seem to help all the time.

**Reply by DTTerastar** [reply]
apple:1005:9-12 means we saw a nearby info message length 9 with a txPower of -12
when we query and get a response from model we then use the model number for the id (apple:iphone13-3)
if you don't have multiple phones that report as apple:1005:9-12 just use that for your automation because it doesn't require an extra step to work.  If you need the more accurate model number to workaround 2 devices with the same id, add apple:1005:9-12 to query and it'll always attempt to query and upgrade the id.

**Reply by harphere** [reply]
Ok. What about the first part of my question where the physical phone was not in the house at all but it reports to my bedroom instance as happened today. In my layman opinion, that would mean there is some other device close by that uses the same id and reports to bedroom which would throw off an automation but the physical phone I want to track was never around.

Sorry if I'm making you repeat what you already explained but I obviously didn't pick that up.

**Reply by DTTerastar** [reply]
A good test of the theory is to check if you see multiple macs flipping back and forth with the same id (in MQTT explorer).  That's what i'm talking about when I say you can't uniquely identify a device.

**Reply by DTTerastar** [reply]
The other 2 ways to get a better id for a phone are:
  * turn on the covid tracing on one phone and use the exp:20 fingerprint.
  * install the room assistant app
